### PR TITLE
Google calendar integration

### PR DESCRIPTION
**Pull request by jboning**
For #132

### CODE REVIEW

File: main/management/commands/auth_gcal.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+from main.lib.gcal import default_token_store, default_oauth_flow
+
+class Command(BaseCommand):
+    def handle(self, *args, **options):
+        print( args, options)
+        store = default_token_store()
+        flow = default_oauth_flow()
+
+        # I haven't figured out how to make oauth2lient.tools's argparser play
```
Comment (ctaylor451) [inline main/management/commands/auth_gcal.py]: You might want to talk with Kevin.  He was working on oauth for the wiki

File: main/views/event_views.py [hunk 2]
```diff
@@ -109,6 +110,13 @@ def get_form(self):
 
         return form
 
+    # Abuse get_success_url to do a calendar update after updating the object.
+    def get_success_url(self, *args, **kwargs):
+        if default_gcal_manager_enabled():
+            default_gcal_manager().sync_event(self.object)
```
Comment (ctaylor451) [inline main/views/event_views.py]: Is there an error handling buried in this?  At a minimum seems like we should post a sentry alert

File: main/management/commands/auth_gcal.py (hunk 1 above)
Comment (jboning) [reply main/management/commands/auth_gcal.py]: I think this library is mainly specific to google. This should be good enough, I don't think it's worth putting any effort into.

File: main/views/event_views.py (hunk 2 above)
Comment (jboning) [reply main/views/event_views.py]: I updated the PR with some code that logs at INFO level if the settings for calendar aren't configured, and ERROR if there's a problem with the credentials.

There's not any explicit error handling for actual requests failing; I believe there will be an uncaught exception, and that should get logged.

**Comment by jboning** [comment]
The ansible vault in this PR contains credentials for the main (bamru.calendar@gmail.com) calendar. That doesn't feel totally right if folks using that configuration for development. But I don't know where else to put them.

**Comment by jboning** [comment]
OK, this now has a token for the bamru.caltest calendar in the ansible vault instead, so it's safe for anyone's dev machine.

**Pull request merged**